allenact/embodiedai/models/visual_nav_models.py

```diff
@@ -0,0 +1,210 @@
+from typing import Callable, Tuple, Dict, Optional, Union, List, cast, Any
+from allenact.utils.system import get_logger
+from collections import OrderedDict
+
+import gym
+import torch
+import torch.nn as nn
+from gym.spaces.dict import Dict as SpaceDict
+
+from allenact.algorithms.onpolicy_sync.policy import (
+    ActorCriticModel,
+    LinearCriticHead,
+    LinearActorHead,
+    ObservationType,
+    DistributionType,
+)
+from allenact.base_abstractions.distributions import CategoricalDistr
+from allenact.base_abstractions.misc import ActorCriticOutput, Memory
+from allenact.utils.model_utils import FeatureEmbedding
+from allenact.embodiedai.models.basic_models import RNNStateEncoder
+from allenact.embodiedai.models.aux_models import AuxiliaryModel
+from allenact.embodiedai.models.fusion_models import FusionModels
+from allenact.embodiedai.aux_losses.losses import TaskWeightsLoss
+
+
+class VisualNavActorCritic(ActorCriticModel[CategoricalDistr]):
+    def __init__(
+        self,
+        action_space: gym.spaces.Discrete,
+        observation_space: SpaceDict,
+        hidden_size=512,
+        multiple_beliefs=False,
+        beliefs_fusion: Optional[str] = None,
```

Comment:
Why not make `beliefs_fusion` an optional type, e.g.:
```
from typing import TypeVar
from allenact.embodiedai.models.fusion_models import Fusion

FusionType = TypeVar("FusionType", bound=Fusion)

class VisualNavActorCritic(ActorCriticModel[CategoricalDistr]):
...
      beliefs_fusion: Optional[FusionType] = None,
...
```
Then you could just do`self.beliefs_fusion(...)` rather than `FusionModels[self.beliefs_fusion]` and we wouldn't have to depend on the `FusionModels` variable containing our fusion method of interest (this would make it easier for others to extend this as well).

Reply:
Very good point. I did not know `TypeVar` has `bound` option.